app/client/src/features/dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from 'react';
+import { graphql, PreloadedQuery, usePreloadedQuery } from 'react-relay';
+import { useRouter } from 'next/router';
+
+import { Button, Card, CardContent, Grid, Link, List, ListItem, Typography, IconButton } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { Add } from '@mui/icons-material';
+
+import { Loader } from '@local/components/Loader';
+import { DashboardEvent } from '@local/features/dashboard/DashboardEvent';
+import { useUser } from '@local/features/accounts';
+import type { DashboardQuery } from '@local/__generated__/DashboardQuery.graphql';
+
+const useStyles = makeStyles((theme) => ({
+    item: {
+        margin: theme.spacing(0, 0, 4, 0),
+    },
+    card: {
+        padding: theme.spacing(1, 1, 1, 1),
+    },
+    title: {
+        marginBottom: theme.spacing(1),
+    },
+    text: {
+        marginLeft: theme.spacing(1),
+    },
+    secondaryText: {
+        color: theme.palette.text.secondary,
+    },
+}));
+
+export const DASHBOARD_QUERY = graphql`
+    query DashboardQuery {
+        me {
+            events {
+                edges {
+                    node {
+                        id
+                        title
+                        description
+                        startDateTime
+                        endDateTime
+                        organization {
+                            name
+                        }
+                    }
+                }
+            }
+        }
+    }
+`;
+
+interface Props {
+    queryRef: PreloadedQuery<DashboardQuery>;
+}
+
+export function Dashboard({ queryRef }: Props) {
+    const data = usePreloadedQuery<DashboardQuery>(DASHBOARD_QUERY, queryRef);
+    const listOfEvents = React.useMemo(() => data.me?.events?.edges ?? [], [data.me]);
+    const classes = useStyles();
+    const router = useRouter();
+    const [user,,isLoading] = useUser();
+    const handleNav = (path: string) => () => router.push(path);
+
+    // Verify user is logged in
+    React.useEffect(() => {
+        if (!isLoading && !user) router.push('/');
+    }, [user, router, isLoading]);
+    
+    if(!data) return <Loader />;
+
+    // Get current date to categorize events as current or upcoming
+    const currentDateTime = new Date();
+
+    // Store current events
+    const currentEvents = listOfEvents.filter(({ node: event }) => {
+        if (event.startDateTime && event.endDateTime) {
+            const eventStartDateTime = new Date(event.startDateTime);
+            const eventEndDateTime = new Date(event.endDateTime);
+            if (eventStartDateTime.getTime() < currentDateTime.getTime() && 
+                eventEndDateTime.getTime() > currentDateTime.getTime()) {
+                return true;
+            }
+        }
+    });
```

Comment:
Ah, one more thing, you'll probably get a lint error since this is technically after a return statement. Just move the return statement lower down. It's not that expensive to calculate these things on the initial render.